### ISSUE TITLE
Dokan returns incorrect WIN32_FIND_DATAA::dwReserved0 after creating a soft connection

### ISSUE THREAD

**Issue by linyuan870** [dokan-dev/dokany#716]
Feature request can skip this form. Bug report **must** complete it. `Check List` must be 100% match or it will be automatically closed without further discussion. Please remove this line.

# Environment
 * Windows version:                   win10
 * Processor architecture:           x64 
 * Dokany version:                     1.1.0.2
 * Library type (Dokany/FUSE):  Dokany
 
# Check List
  - [x] I checked my issue [doesn't exist yet](https://github.com/dokan-dev/dokany/issues?utf8=%E2%9C%93&q=is%3Aissue)
  - [x] My issue is valid with mirror default sample and not specific to my user-mode driver implementation
  - [x] I can always reproduce the issue with the provided description below.
  - [x] I have updated Dokany to the [latest version](https://github.com/dokan-dev/dokany/releases) and have reboot my computer after.
  - [x] I tested one of the last [snapshot](https://github.com/dokan-dev/dokany/wiki/Build#user-snapshot) from appveyor CI

**Comment by Liryna** [comment]
Hi @linyuan870 ,

Could you explain a little more the issue ?

**Comment by linyuan870** [comment]
@Liryna 
https://docs.microsoft.com/en-us/windows/desktop/FileIO/determining-whether-a-directory-is-a-volume-mount-point
**In the windows system disk such as G:, create a symbolic link softlink to a file test.txt**
**The symbolic link type created is .symlink.**
![10](https://user-images.githubusercontent.com/11846535/42486059-e5704362-842c-11e8-8cc2-ad262bcc4094.jpg)
![11](https://user-images.githubusercontent.com/11846535/42486061-e5c7632c-842c-11e8-8695-49d39649a63b.jpg)

**But when the mirror creates a symbolic link, its type is not .symlink.**
**Use the FindFirstFile and FindNextFile functions to obtain the reparse tag in the dwReserved0 member of the WIN32_FIND_DATA structure.**
**In the right case, dwReserved0 should be set to IO_REPARSE_TAG_SYMLINK. But the value in WIN32_FIND_DATA is not this. Is it where Dokan does not handle this structure?**
![00](https://user-images.githubusercontent.com/11846535/42486613-9e3b716c-842f-11e8-847a-12203879b00d.png)

**Comment by linyuan870** [comment]
@Liryna I think I didn’t make it clear last time.

When I use mirror.exe **/r G: /l m**, the symlink type originally located on the G drive has changed(Changed from .symlink to file); I saw in the mirror's MirrorFindFiles function that the value of the dwReserved0 of the WIN32_FIND_DATAW structure has changed; windows passed over dwReserved0 is set to IO_REPARSE_TAG_SYMLINK, the value returned after dokan has changed; it should be where the dokan changes the structure, resulting in the mirror disk can not display the .symlink file like the G drive.

![capture](https://user-images.githubusercontent.com/11846535/42855921-5874c20e-8a75-11e8-827a-e02f6f6d0e03.PNG)

**Comment by Liryna** [comment]
Hi @linyuan870,

`IO_REPARSE_TAG_SYMLINK` is part of the reparse point feature currently not supported by Dokan.
With the support of this feature, dwReserved0 will be correctly set with the correct flags.

**Comment by linyuan870** [comment]
@Liryna 
**dokan/directory.c**
function DokanFillIdBothDirInfo
Buffer->EaSize = 0;    =》 Buffer->EaSize = FindData->dwReserved0;
function DokanFillBothDirInfo
Buffer->EaSize = 0;   =》Buffer->EaSize = FindData->dwReserved0;
**After the above changes, type can correctly display .symlink.**
But still can't achieve the function of soft connection, it should be supported and should be supported. I hope someone can provide this support.

**Comment by Liryna** [comment]
How did you found this ? `EaSize` is for extended attributes.

**Comment by linyuan870** [comment]
I think the type error is to throw dwReserved0 when it is done through dokan.

**Comment by Liryna** [comment]
Yes, I tried to found also but could not. Probably it is really linked to the repase point feature. The kernel probably link both information to create dwReserved0.
